From 8627d178589db9888f121b5b13e7049f46e9d0f6 Mon Sep 17 00:00:00 2001
From: Jan Petrous <jan.petrous@nxp.com>
Date: Fri, 24 Feb 2023 10:07:37 +0100
Subject: [PATCH 60/63] arm64: dts: s32g-pfe: introduce nxp,pfeng-linked-phyif
 ethernet property

To support additional 'remote' HIF channels (CPU ports)
we need to extend possibility of link netdev not only
with EMAC ports, but also with HIF channels which
are not driven by the driver instance. Those HIF channels
could be treated as remote CPU ports, owned by another
driver instance.

The original property nxp,pfeng-emac-id is replaced by
nxp,pfeng-linked-phyif to support those remote HIF ports.

Issue: AAVB-6542

Upstream-Status: Pending 

Signed-off-by: Jan Petrous <jan.petrous@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm64/boot/dts/freescale/s32g-pfe-slave.dtsi | 8 ++++----
 arch/arm64/boot/dts/freescale/s32g-pfe.dtsi       | 6 +++---
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/s32g-pfe-slave.dtsi b/arch/arm64/boot/dts/freescale/s32g-pfe-slave.dtsi
index 4b5e64466d60..e40e24c55e0e 100644
--- a/arch/arm64/boot/dts/freescale/s32g-pfe-slave.dtsi
+++ b/arch/arm64/boot/dts/freescale/s32g-pfe-slave.dtsi
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0
 /*
- * Copyright 2022 NXP
+ * Copyright 2022-2023 NXP
  *
  * Net devices mapping:
  *   pfe0sl: to EMAC_0
@@ -42,7 +42,7 @@ pfesl_netif0: ethernet@100 {
 				local-mac-address = [ 00 04 9F BE FF 00 ];
 				nxp,pfeng-if-name = "pfe0sl";
 				nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_3>;
-				nxp,pfeng-emac-id = <PFE_EMAC_0>;
+				nxp,pfeng-linked-phyif = <PFE_PHYIF_EMAC_0>;
 			};
 
 			/* Network interface 'pfe1sl' */
@@ -53,7 +53,7 @@ pfesl_netif1: ethernet@101 {
 				local-mac-address = [ 00 04 9F BE FF 01 ];
 				nxp,pfeng-if-name = "pfe1sl";
 				nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_3>;
-				nxp,pfeng-emac-id = <PFE_EMAC_1>;
+				nxp,pfeng-linked-phyif = <PFE_PHYIF_EMAC_1>;
 			};
 
 			/* Network interface 'pfe2sl' */
@@ -64,7 +64,7 @@ pfesl_netif2: ethernet@102 {
 				local-mac-address = [ 00 04 9F BE FF 02 ];
 				nxp,pfeng-if-name = "pfe2sl";
 				nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_3>;
-				nxp,pfeng-emac-id = <PFE_EMAC_2>;
+				nxp,pfeng-linked-phyif = <PFE_PHYIF_EMAC_2>;
 			};
 
 			/* Network interface 'aux0sl' */
diff --git a/arch/arm64/boot/dts/freescale/s32g-pfe.dtsi b/arch/arm64/boot/dts/freescale/s32g-pfe.dtsi
index a49da41be800..14fdb47f4828 100644
--- a/arch/arm64/boot/dts/freescale/s32g-pfe.dtsi
+++ b/arch/arm64/boot/dts/freescale/s32g-pfe.dtsi
@@ -88,7 +88,7 @@ pfe_netif0: ethernet@10 {
 				local-mac-address = [ 00 04 9F BE EF 00 ];
 				nxp,pfeng-if-name = "pfe0";
 				nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_0>;
-				nxp,pfeng-emac-id = <PFE_EMAC_0>;
+				nxp,pfeng-linked-phyif = <PFE_PHYIF_EMAC_0>;
 				clocks = <&clks S32G_SCMI_CLK_PFE0_TX_SGMII>,
 					 <&clks S32G_SCMI_CLK_PFE0_TX_RGMII>,
 					 <&clks S32G_SCMI_CLK_PFE0_TX_RMII>,
@@ -111,7 +111,7 @@ pfe_netif1: ethernet@11 {
 				local-mac-address = [ 00 04 9F BE EF 01 ];
 				nxp,pfeng-if-name = "pfe1";
 				nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_1>;
-				nxp,pfeng-emac-id = <PFE_EMAC_1>;
+				nxp,pfeng-linked-phyif = <PFE_PHYIF_EMAC_1>;
 				clocks = <&clks S32G_SCMI_CLK_PFE1_TX_SGMII>,
 					 <&clks S32G_SCMI_CLK_PFE1_TX_RGMII>,
 					 <&clks S32G_SCMI_CLK_PFE1_TX_RMII>,
@@ -134,7 +134,7 @@ pfe_netif2: ethernet@12 {
 				local-mac-address = [ 00 04 9F BE EF 02 ];
 				nxp,pfeng-if-name = "pfe2";
 				nxp,pfeng-hif-channels = <PFE_HIF_CHANNEL_2>;
-				nxp,pfeng-emac-id = <PFE_EMAC_2>;
+				nxp,pfeng-linked-phyif = <PFE_PHYIF_EMAC_2>;
 				clocks = <&clks S32G_SCMI_CLK_PFE2_TX_SGMII>,
 					 <&clks S32G_SCMI_CLK_PFE2_TX_RGMII>,
 					 <&clks S32G_SCMI_CLK_PFE2_TX_RMII>,
-- 
2.25.1

